src/components/ItemList/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import { updateItemDate } from '../Utils/firestore';
+import { differenceInMilliseconds } from 'date-fns';
+
+function ItemList({ itemName, docId, lastPurchasedDate }) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [token] = useState(() => window.localStorage.getItem('tcl18-token'));
+
+  const MarkProductPurshased = (id) => {
+    updateItemDate(token, id)
+      .then(() => {
+        console.log('product updated');
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  useEffect(() => {
+    if (lastPurchasedDate && isDisabled === false) {
+      const hoursAtTheMoment = differenceInMilliseconds(
```

Comment:
I'd consider not using a library for this. You can get the difference in milliseconds using native javascript, here's a quick comparison between libs: https://github.com/you-dont-need/You-Dont-Need-Momentjs#difference

As a general rule, anything you can do natively will mean a smaller bundle size, which makes your apps faster (or so I've been told, maybe @alejandronanez or @indiragp have more experience with optimization here!). Libs are great when you'll be using them a lot all over the place so since you're importing just 1 method for now, maybe you don't need to :)

Reply:
That part of code was refactored.